### PR TITLE
CheckoutSessions — top level Setup Future Usage

### PR DESCRIPTION
**Pull request by gbirch-stripe**
## Summary
Implement top level SFU. PMO SFU in subsequent PR

## Motivation
https://docs.google.com/document/d/1CWglcOHR6I8cOWKjztPfnhnVG8axucpv-j5cWIoey7E/edit?pli=1&tab=t.g4karprubz63#heading=h.qr0ujyg56e8u

## Testing
- unit tests
- API mock tests
- payment sheet form factory tests

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Checkout/API Models/STPCheckoutSession.swift [hunk 1]
```diff
@@ -458,8 +465,12 @@ extension STPCheckoutSession: STPAPIResponseDecodable {
 // MARK: - Parsing Helpers
 
 extension STPCheckoutSession {
-    func merchantWillSavePaymentMethod(_ paymentMethodType: STPPaymentMethodType) -> Bool {
+    func setupFutureUsage(for paymentMethodType: STPPaymentMethodType) -> String? {
         _ = paymentMethodType
+        return setupFutureUsage
+    }
```
Comment (gbirch-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Checkout/API Models/STPCheckoutSession.swift:471]: PMO SFU logic will go here

**Comment by gbirch-stripe** [comment]
verify public interface is acting up 😞 . verified that no public api changes are being made

File: StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/Checkout/STPCheckoutSessionTest.swift [hunk 2]
```diff
@@ -339,6 +362,21 @@ class STPCheckoutSessionTest: XCTestCase {
         XCTAssertFalse(session.merchantWillSavePaymentMethod(.card))
     }
 
+    func testMerchantWillSavePaymentMethod_paymentModeWithTopLevelSetupFutureUsage() {
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/Checkout/STPCheckoutSessionTest.swift:365]: Is it worth adding a test for `none` `setup_future_usage` value?

**Review by porter-stripe — COMMENTED** [review]
looks good one question

**Review by porter-stripe — APPROVED** [review]
